cpu/msp430/mtarch.c

```diff
@@ -66,10 +66,10 @@ mtarch_start(struct mtarch_thread *t,
 
   t->sp = &t->stack[MTARCH_STACKSIZE - 1];
 
-  *t->sp = (unsigned short)mt_exit;
+  *t->sp = (unsigned short)(void *)mt_exit;
```

Comment:
Actually the original warning is pretty clear : <pre>warning: cast from pointer to integer of different size</pre>
I just tested :
<pre>
  printf("%d\n", sizeof(int *));
  printf("%d\n", sizeof(testfunc));
</pre>
gives : 
<pre>
2
4
</pre>
A pointer to a function will be 16bits only if the compiler knows that the function is in the near region. But there is no issue because mtarch is not used in vanilla Contiki.

@simonduq void * can only be used reliably for data pointer, not for function pointer